config/confighttp/confighttp.go

```diff
@@ -216,8 +216,13 @@ func (hcs *ClientConfig) ToClient(ctx context.Context, host component.Host, sett
 
 	// Compress the body using specified compression methods if non-empty string is provided.
 	// Supporting gzip, zlib, deflate, snappy, and zstd; none is treated as uncompressed.
+	var compressionTypeWithLevel configcompression.TypeWithLevel
+	err = compressionTypeWithLevel.UnmarshalText([]byte(hcs.Compression))
```

Comment:
We don't need to call UnmarshalText explicitly. Changing `ClientConfig.Compression` field from `configcompression.Type` to `configcompression.TypeWithLevel` should be enough

Reply:
Doing that seemed to be a breaking change. There are errors like this seen in multiple contrib components

```Error: config/config.go:147:24: invalid operation: cfg.Compression != "" (mismatched types configcompression.TypeWithLevel and untyped string)```

Reply:
Yes, it is a Go API breaking change, but not an end-user breaking change. I think this is the cleanest solution. @open-telemetry/collector-approvers let me know if you have other ideas

Reply:
if we make the API breaking change, are going to mute all the failing tests (in this repo) and have the contrib component owners fix that later. Sorry, I'm not clear on this process.

Reply:
Right. If we go this route, we would need to quickly submit a follow-up fix in the contrib repo

Reply:
How about keeping this as is (non-breaking) for this PR? and handle all breaking changes (core and contrib) in  2 other PRs. The only reason is, they seem to affect multiple contrib components (almost all)

Reply:
@dmitryax I have pushed the API breaking changes and the contrib tests(`contrib-tests / contrib-tests-matrix`) seem to be failing.

Reply:
If you have an idea how to make it in several steps without breaking contrib and get this result, that would be great. I'm not aware of any

Reply:
I think we can stick to the current plan. These are the only changes that need to go into the contrib repository

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35843/files